File: 11_container_with_most_water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i = 0
        j = len(height)-1
        c = []
        while(i<j):
            if height[i]<height[j]:
                
                d = height[i]*(j-i)
                c.append(d)
                i+=1
            else:
                d = height[j]*(j-i)
                c.append(d)
                j-=1

        c.sort()
        return max(c)

File: 11_container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& v) {
        if(v.size()==2)
            return v[0]<=v[1]?v[0]:v[1];
        int start=0, end=v.size()-1;
       int ans=0;
        while(start<end)
        {
            ans=max(ans,(min(v[start],v[end])*(end-start)));
            if(v[start]<=v[end]||v[start+1]>=v[start]+1)
                start++;
            else if(v[end]<=v[start]||v[end-1]>=v[end]+1)
                end--;
            
        }
        
        return ans;
    }
};